### PR TITLE
LPS-134492 Add language selector to blueprints

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-134492 - Blueprint title and description are localizable through the UI using a language selector

Some notes I learned along the way:
- Add `'X-Liferay-Accept-All-Languages': true` to the header to see the title and description map in the response.
- Throughout the editor, the title and description are now saved in BCP47 form (like `en-US`) since portal uses that form for the language selector. However, they are submitted in the general `en_US` form, as expected for backend. Utils `formatLocaleWithUnderscores` and `formatLocaleWithDashes` have been added to make it easier to convert back and forth.
- The Chinese locales `zh-Hans-CN` and `zh-Hant-TW` need to be adapted to `zh-CN` and `zh-TW` in order to be viewable on language selector and submittable.
- Language selector is based off `Liferay.Language.available`. If a blueprint has certain titles/descriptions that are not part of availableLanguages, they do not show up in the language selector (backend filters them out for the getSXPBlueprint response) and they would not be saved.

Let me know if there's any questions/things to add! Thanks!

### CODE REVIEW

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/PageToolbar.js [hunk 1]
```diff
@@ -50,67 +128,81 @@ function EditTitleModal({
...
 							autoFocus={modalFieldFocus === 'title'}
 							id="title"
-							onBlur={({currentTarget}) => {
-								setHasError(!currentTarget.value);
-							}}
-							onChange={({target: {value}}) => setTitle(value)}
+							label={_getTitleLabel()}
+							locales={localesForSelector}
+							onBlur={_handleBlur}
+							onSelectedLocaleChange={_handleChangeLocale(
```
Comment (thektan) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/PageToolbar.js:163]: Match handler name with prop. In this case it should be `_handleSelectedLocaleChange`.

Let's also follow the pattern of `_handle*Change`, where specific names go after `_handle` so the handler functions are grouped together by component (if there are multiple handlers like `_handle*Blur`, `_handle*Focus`).

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/PageToolbar.js [hunk 2]
```diff
@@ -50,67 +128,81 @@ function EditTitleModal({
...
 							component="textarea"
 							id="description"
-							onChange={({target: {value}}) =>
-								setDescription(value)
-							}
-							type="text"
-							value={description}
+							label={Liferay.Language.get('description')}
+							locales={localesForSelector}
+							onSelectedLocaleChange={_handleChangeLocale(
```
Comment (thektan) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/PageToolbar.js:196]: Same as previous comment. `_handleSelectedLocaleChange`

**Review by thektan — CHANGES_REQUESTED** [review]
Those util functions look good! I'm requesting a few formatting changes. Overall good work on this though!

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/PageToolbar.js (hunk 1 above)
Comment (oliv-yu) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/PageToolbar.js]: Added, thanks! `_handle*Change` got it 👍

**Review by thektan — APPROVED** [review]
Thanks @oliv-yu!

**Comment by thektan** [comment]
ci:forward

**Pull request closed without merging**